tests/test_pretty_midi.py

```diff
@@ -272,3 +273,59 @@ def simple():
     for ks, t, k in zip(pm.key_signature_changes, ks_times, ks_keys):
         assert ks.time == t
         assert ks.key_number == k
+
+
+def test_properly_order_overlapping_notes():
+    def make_mido_track(notes_str, path):
+        track = mido.MidiTrack()
+        for line in notes_str.split('\n'):
+            line = line.strip()
+            if line:
+                track.append(mido.Message.from_str(line))
+        mido_file = mido.MidiFile()
+        mido_file.tracks.append(track)
+        mido_file.save(path)
+
+    # two notes with pitch 72 open at once
+    bad_track = """
+    note_on channel=0 note=72 velocity=88 time=0
+    note_on channel=0 note=72 velocity=88 time=48
+    note_on channel=0 note=72 velocity=0 time=0
+    note_on channel=0 note=74 velocity=88 time=48
+    note_on channel=0 note=72 velocity=0 time=0
+    note_on channel=0 note=72 velocity=88 time=48
+    note_on channel=0 note=74 velocity=0 time=0
+    note_on channel=0 note=72 velocity=0 time=48
+    """
+
+    # the 72 note is first closed, then another one opened
+    good_track = """
+    note_on channel=0 note=72 velocity=88 time=0
+    note_on channel=0 note=72 velocity=0 time=48
+    note_on channel=0 note=72 velocity=88 time=0
+    note_on channel=0 note=74 velocity=88 time=48
+    note_on channel=0 note=72 velocity=0 time=0
+    note_on channel=0 note=72 velocity=88 time=48
+    note_on channel=0 note=74 velocity=0 time=0
+    note_on channel=0 note=72 velocity=0 time=48
+    """
+
+    for kind, track in (('good', good_track), ('bad', bad_track)):
+        midi_path = 'overlapping_notes_%s_track.mid' % kind
+        make_mido_track(track, midi_path)
+
+        pm_song = pretty_midi.PrettyMIDI(midi_path)
+
+        def extract_notes(pm_track):
+            return np.array([(note.pitch, note.end-note.start) for note
+                             in pm_track.notes])
+
+        expected = np.array([[72, 0.05], [72, 0.05], [74, 0.05], [72, 0.05]])
+        assert np.allclose(expected, extract_notes(pm_song.instruments[0]))
+
+        midi_written_path = 'overlapping_notes_%s_track_written.mid' % kind
```

Comment:
Yes. Fixed in 83e8ebd. We reuse the NamedTemporaryFile while is open using `file.seek(0)`, otherwise on `close()` it gets deleted.